lib/src/routing/connection_registry.rs

```diff
@@ -1,350 +1,221 @@
-                    .get_mut(db_name.as_str())
-                    .unwrap()
-                    .remove(index);
-                self.pool_registry.remove(server);
-            } else {
-                debug!("Server not found in the registry: {:?}", server);
+                Ok(new_table) => {
+                    let db = new_table.db.clone();
+                    let database_table: DatabaseTable = new_table.into();
+                    let servers = database_table.resolve();
+                    let db_name = db.map_or(String::new(), |d| d.to_string());
+                    debug!("Routing table for database {db_name} refreshed");
+                    servers
+                }
+                Err(e) => {
+                    error!("Failed to refresh routing table for database default: {e}");
+                    vec![]
+                }
             }
         }
     }
 
-    /// Get all available Bolt servers for a specific database or the default database if none is provided.
-    pub fn servers(&self, db: Option<Database>) -> Vec<BoltServer> {
-        let db_name = self.get_db_name(db);
-        if self.databases.contains_key(db_name.as_str()) {
-            self.databases
-                .get(db_name.as_str())
-                .map(|entry| entry.value().clone())
-                .unwrap_or_default()
-        } else {
-            debug!("Creating new registry for database: {}", db_name);
-            self.databases.insert(db_name.clone(), Vec::new());
-            vec![]
+    pub fn all_servers(&self) -> Vec<BoltServer> {
+        self.databases
+            .values()
+            .iter()
+            .flat_map(|dt| dt.resolve())
+            .collect()
+    }
+
+    pub fn update(&self, config: &Config, routing_table: &RoutingTable) -> Result<(), Error> {
+        let servers = routing_table.resolve();
+
+        for server in servers.iter() {
+            let neo_url = server.to_neo_url(self.scheme.as_str());
+            if self.pool_registry.contains_key(&neo_url) {
+                debug!("Server already exists in the registry: {server:?}");
+                continue;
+            }
+            let uri = format!("{}://{}:{}", self.scheme, server.address, server.port);
+            debug!("Creating pool for server: {uri}");
+            self.pool_registry.insert(
+                neo_url,
+                create_pool(&Config {
+                    uri,
+                    ..config.clone()
+                })?,
+            );
         }
+        let db_name = routing_table
+            .db
+            .as_ref()
+            .map_or("".to_string(), |d| d.to_string());
+        debug!(
+            "Registry updated for database {}. New size is {} with TTL {}s",
+            db_name,
+            self.pool_registry.len(),
+            routing_table.ttl
+        );
+        let database_table: DatabaseTable = routing_table.into();
+        self.databases.insert(db_name.clone(), database_table);
```

Comment:
Also can avoid a to_string here

```suggestion
        let db_name = routing_table.db.as_deref().unwrap_or("");
        debug!(
            "Registry updated for database {}. New size is {} with TTL {}s",
            db_name,
            self.pool_registry.len(),
            routing_table.ttl
        );
        let database_table: DatabaseTable = routing_table.into();
        self.databases.insert(db_name.to_owned(), database_table);
```